src/_pytest/cacheprovider.py

```diff
@@ -179,16 +179,26 @@ def set(self, key: str, value: object) -> None:
             else:
                 cache_dir_exists_already = self._cachedir.exists()
                 path.parent.mkdir(exist_ok=True, parents=True)
-        except OSError:
-            self.warn("could not create cache path {path}", path=path, _ispytest=True)
+        except OSError as exc:
+            self.warn(
+                "could not create cache path {path}: {exc}",
+                path=path,
+                exc=exc,
+                _ispytest=True,
+            )
```

Comment:
```suggestion
            self.warn(
                f"could not create cache path {path}: {exc}",
                _ispytest=True,
            )
```

Reply:
If we make the suggested changes, should the `warn()` implementation be simplified as well to not take fmt+kwargs but the ready formatted message? These two are the only callers that I can see, and `warn()` is marked `:private:` in the docs.

Reply:
> If we make the suggested changes, should the warn() implementation be simplified as well to not take fmt+kwargs but the ready formatted message? 

I like the idea, but is probably best to leave it alone given it is public: might there be code in the wild which depends on it.